src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GenerateVCFFromPosteriors.java

```diff
@@ -0,0 +1,224 @@
+            fullName = SAMPLE_DIRECTORY_NAME_FULL_NAME,
+            shortName = SAMPLE_DIRECTORY_NAME_SHORT_NAME
+    )
+    private String sampleDirectoryName = null;
+
+    @Argument(
+            doc = "Output VCF file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+    )
+    private File outputFile = null;
+
+    /**
+     * Sample name derived from the copy number posterior file
+     */
+    private String sampleName;
+
+    final String variantPrefix = "CNV";
+
+    @Override
+    public void traverse() {
+        if (orderedChunkDirectoryList.size() <= 0) {
+            throw new UserException.BadInput("You must specify at least one chunk directory");
+        }
+
+        final SimpleIntervalCollection firstChunkSimpleIntervalCollection =
+                new SimpleIntervalCollection(new File(getIntervalFileFromChunkDirectory(orderedChunkDirectoryList.get(0).getAbsolutePath()).getAbsolutePath()));
+        final SAMSequenceDictionary samSequenceDictionary = firstChunkSimpleIntervalCollection.getMetadata().getSequenceDictionary();
+        Triple<List<CopyNumberPosteriorLocatableRecord>, IntegerCopyNumberStateCollection, String> firstCopyNumberPosteriorFileInfo =
+                readChunkedPosteriorFileFromDirectory(orderedChunkDirectoryList.get(0), samSequenceDictionary);
+        final IntegerCopyNumberStateCollection copyNumberStateCollection = firstCopyNumberPosteriorFileInfo.getMiddle();
+        sampleName = firstCopyNumberPosteriorFileInfo.getRight();
+
+        if (firstCopyNumberPosteriorFileInfo.getMiddle().size() < 3) {
+            throw new UserException.BadInput("There should be at least 3 copy number states");
+        }
+
+        final VariantContextWriter outputWriter = GATKVariantContextUtils.createVCFWriter(outputFile, null, false);
+        final GCNVPostProcessor gcnvPostProcessor = new GCNVPostProcessor(outputWriter, copyNumberStateCollection,
+                sampleName, samSequenceDictionary);
+
+        //TODO pass the command line invocation string to the header composer method
+        gcnvPostProcessor.composeVariantContextHeader(null);
+        int currentChunk = 0;
+
+        for(File chunkRootDirectory: orderedChunkDirectoryList) {
+            currentChunk++;
+            logger.info(String.format("Analyzing copy number posterior chunk %d", currentChunk));
+
+            Triple<List<CopyNumberPosteriorLocatableRecord>, IntegerCopyNumberStateCollection, String> chunkedCopyNumberPosteriorFileInfo =
+            readChunkedPosteriorFileFromDirectory(chunkRootDirectory, samSequenceDictionary);
+            //check that the copy number state collection is equivalent across all chunks
+            if (!copyNumberStateCollection.equals(chunkedCopyNumberPosteriorFileInfo.getMiddle())) {
+                throw new UserException.BadInput("Copy number collection differs across chunked posterior outputs");
```

Comment:
periods here and in other `BadInput` messages throughout

Reply:
Done